picard/file.py

```diff
@@ -780,20 +776,26 @@ def _info(self, metadata, file):
             metadata['~format'] = self.NAME
         else:
             metadata['~format'] = self.__class__.__name__.replace('File', '')
-        self._add_path_to_metadata(metadata)
+        self._update_filesystem_metadata(metadata)
 
-    def _add_path_to_metadata(self, metadata):
-        metadata['~dirname'] = os.path.dirname(self.filename)
-        filename, extension = os.path.splitext(os.path.basename(self.filename))
-        metadata['~filename'] = filename
+    def _update_filesystem_metadata(self, metadata):
+        filename = encode_filename(self.filename)
+        metadata['~dirname'] = os.path.dirname(filename)
+        filename_no_ext, extension = os.path.splitext(os.path.basename(filename))
+        metadata['~filename'] = filename_no_ext
         metadata['~extension'] = extension.lower()[1:]
 
         try:
-            created = os.path.getctime(self.filename)
+            metadata['~filesize'] = os.path.getsize(filename)
+        except BaseException as e:
```

Comment:
```suggestion
        except OSError as e:
```

Reply:
Or, just move the filesize extraction inside the try/except block below, and make the error message a bit more generic

Reply:
I updated the exception handling, but kept it separate from the timestamp case

Reply:
> I updated the exception handling, but kept it separate from the timestamp case

I don't see any good reason to keep it separated, because they all raise OSError, a more generic error message can do, plus if we can't get size, it is very likely we can't get ctime/mtime from the same file.